### PR TITLE
docs(typedoc): Setup TypeDoc plus docs improvements

### PR DESCRIPTION
**Pull request by JoseLion**
This PR adds [TypeDoc](https://typedoc.org/) to the code base so we can automatically generate an API Reference based on JSDoc comments spread throughout the code. The changes on this PR include:
- A few improvements to the code which were necessary to resolve all TypeDoc warnings
- Some changes in GH Pages and GH Actions to include the auto-generated API Reference.
- The `README.md` file was improved to show only the library usage and redirect to the API Reference for more details on types and the complete API, reducing the maintenance cost of the documentation.
- Added two more GIFs to show the new motion effects.

### CODE REVIEW

File: .github/workflows/pages.yml [hunk 1]
```diff
@@ -18,17 +18,19 @@ jobs:
...
+      - run: yarn install --immutable
+      - run: yarn docs
+      - uses: actions/configure-pages@v2
+      - uses: actions/jekyll-build-pages@v1
         with:
           source: ./
           destination: ./_site
-      - name: Upload artifact
-        uses: actions/upload-pages-artifact@v1
+      - uses: actions/upload-pages-artifact@v1
```
Comment (CristhianMotoche) [inline .github/workflows/pages.yml:33]: @JoseLion I noticed you removed the `name` property from these steps, any reason for doing that?

File: README.md [hunk 2]
```diff
@@ -2,23 +2,30 @@
 
 [![CI](https://github.com/stackbuilders/react-native-spotlight-tour/actions/workflows/ci.yml/badge.svg)](https://github.com/stackbuilders/react-native-spotlight-tour/actions/workflows/ci.yml)
 [![Release](https://github.com/stackbuilders/react-native-spotlight-tour/actions/workflows/release.yml/badge.svg)](https://github.com/stackbuilders/react-native-spotlight-tour/actions/workflows/release.yml)
+[![NPM version](https://img.shields.io/npm/v/@stackbuilders/react-native-spotlight-tour)](https://www.npmjs.com/package/@stackbuilders/react-native-spotlight-tour)
+[![NPM downloads](https://img.shields.io/npm/dm/@stackbuilders/react-native-spotlight-tour)](https://www.npmjs.com/package/@stackbuilders/react-native-spotlight-tour)
+[![NPM license](https://img.shields.io/npm/l/@stackbuilders/react-native-spotlight-tour)](./LICENSE)
+[![GitHub Release Date](https://img.shields.io/github/release-date/stackbuilders/react-native-spotlight-tour)](https://github.com/stackbuilders/react-native-spotlight-tour/releases)
+[![Snyk Vulnerabilities](https://img.shields.io/snyk/vulnerabilities/npm/@stackbuilders/react-native-spotlight-tour)](https://snyk.io/)
```
Comment (CristhianMotoche) [inline README.md:9]: Nice!

File: typedoc.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  "cleanOutputDir": true,
+  "entryPoints": ["src/index.ts"],
+  "entryPointStrategy": "expand",
+  "gitRevision": "release",
+  "githubPages": false,
+  "hideGenerator": true,
+  "includeVersion": false,
+  "mergeModulesMergeMode": "project",
+  "mergeModulesRenameDefaults": true,
+  "name": "AssertiveTS - API Reference",
```
Comment (CristhianMotoche) [inline typedoc.json:12]:
```suggestion
  "name": "React Native Spotlight Tour - API Reference",
```
?

**Review by CristhianMotoche — COMMENTED** [review]
@JoseLion LGTM! I left a question an a suggestion. Let me know your thoughts.

File: .github/workflows/pages.yml (hunk 1 above)
Comment (JoseLion) [reply .github/workflows/pages.yml:33]: Only for the sake of keeping consistency with the other workflow `.yml` files. I do prefer having names on the step, but we can do that for all workflows on a separate PR to maintain consistency 🙂

File: typedoc.json (hunk 3 above)
Comment (JoseLion) [reply typedoc.json:12]: Great catch! Thank you @CristhianMotoche 🎉

File: .github/workflows/pages.yml (hunk 1 above)
Comment (CristhianMotoche) [reply .github/workflows/pages.yml:33]: Sounds good!

**Review by CristhianMotoche — APPROVED** [review]
:shipit:

**Comment by beKoool** [comment]
Hey @JoseLion , The link to API Reference says it does not exist. Can you explain why does it not work? 🤔
Link: https://stackbuilders.github.io/react-native-spotlight-tour/docs/build
image

**Pull request merged**